tfjs-core/src/util_base.ts

```diff
@@ -749,3 +754,38 @@ export function isPromise(object: any): object is Promise<unknown> {
   //  pre-patched Promise.
   return object && object.then && typeof object.then === 'function';
 }
+
+if (!('setTimeoutWPM' in window)) {
+  const messageName = 'setTimeoutWPM';
+  let fns: Function[] = [];
+  let handledMessageCount = 0;
+
+  // If the setTimeout nesting level is greater than 5 and timeout is less than
+  // 4ms, timeout will be clamped to 4ms, which hurts the perf. Interleaving
+  // window.postMessage and setTimeout will trick the browser and avoid the
+  // clamp.
+  const setTimeoutWPM = function(fn: Function, timeout: number) {
+    fns.push(fn);
+    setTimeout(() => {
+      window.postMessage({name: messageName, index: fns.length - 1}, '*');
+    }, timeout);
+  };
+
+  const handleMessage = function(event: MessageEvent) {
+    if (event.source == window && event.data.name == messageName) {
+      event.stopPropagation();
+      const fn = fns[event.data.index];
+      fn();
+      handledMessageCount++;
+      // console.log(`handledMessageCount=${handledMessageCount},
+      // fns.length=${fns.length}`);
+      if (handledMessageCount === fns.length) {
+        fns = [];
+        handledMessageCount = 0;
+      }
+    }
+  };
+
+  window.addEventListener('message', handleMessage, true);
+  (window as any).setTimeoutWPM = setTimeoutWPM;
```

Comment:
It may be better to store this in `_tfGlobals` (instead of `window`), which you can get by calling `getGlobalNamespace()` from [global_util.ts](https://github.com/tensorflow/tfjs/blob/master/tfjs-core/src/global_util.ts#L24).

Reply:
Actually, since this relies on `window`, _tfGlobals may be the wrong place for it. Maybe we should add an optional `setTimeoutWPM` function to [platform.ts](https://github.com/tensorflow/tfjs/blob/master/tfjs-core/src/platforms/platform.ts) and add the function itself to [platform_browser.ts](https://github.com/tensorflow/tfjs/blob/master/tfjs-core/src/platforms/platform_browser.ts) (like Ping suggested in his review).

Reply:
Moved to platform_browser.ts